src/Domain/Service/NoticeAssembler.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace App\Domain\Service;
+
+use App\DTO\Contribution;
+use App\Entity\Contributor;
+use App\Entity\DomainName;
+use App\Entity\MatchingContext;
+use App\Entity\Notice;
+use App\Helper\NoticeVisibility;
+use App\Repository\ContributorRepository;
+use App\Repository\DomainNameRepository;
+use InvalidArgumentException;
+
+class NoticeAssembler
+{
+    private $contributorRepository;
+    private $domainNameRepository;
+
+    public function __construct(ContributorRepository $contributorRepository, DomainNameRepository $domainNameRepository)
+    {
+        $this->contributorRepository = $contributorRepository;
+        $this->domainNameRepository = $domainNameRepository;
+    }
+
+    public function assemble(Contribution $contribution): Notice
+    {
+        if ($contribution->isAQuestion()) {
+            $questionedContributor = $this->contributorRepository->find($contribution->getToContributorId());
+
+            if (!$questionedContributor) {
+                throw new InvalidArgumentException("Questioned contributor doesn't exist.");
+            }
+        }
+
+        $contributor = self::assembleContributor($contribution);
+
+        // @todo /!\ Nothing here prevent a user to impersonate another one /!\
+        $existingContributor = $this->contributorRepository->findOneBy(['email' => $contributor->getEmail()]);
+        if ($existingContributor) {
+            $contributor = $existingContributor;
+        }
+
+        $matchingContext = self::assembleMatchingContext($contribution);
+
+        return self::assembleNotice($contribution)
+            ->setContributor($contributor)
+            ->addMatchingContext($matchingContext)
+        ;
+    }
+
+    public static function assembleContributor(Contribution $contribution): Contributor
+    {
+        return (new Contributor())
+            ->setName($contribution->getContributorName())
+            ->setEmail($contribution->getContributorEmail())
+            ->setEnabled(false)
```

Comment:
I don't know you tell me! Ok good to know, I'll make it `enabled` by default. Any other considerations on a contributor?
I wonder what it'll take if an admin want to allow this contributor to login to the system then...

Reply:
What do @MaartenLMEM @christpet think here ? Should a self-created contributor be initially enabled or disabled ?

Reply:
>  Should a self-created **contributor** be initially enabled or disabled ?
Contributor can be enabled. I don't see any reason I shouldn't, because contributor need minimum one published notice to appear in public contributors list.

- _Then_ Should a self-created **contribution** be initially published or private ?
If it"s directly published, there will be some mistakes sometimes, but doesn't matter, nobody follows in the beginning.

Reply:
@MaartenLMEM we introduced to new flags :
- `draft` statuts
- `question` statuts

Both are different `public`, so they aren't visible as long as the visibility status isn't switch to `public` by a member of the contribution team.